### PR TITLE
Up19 10 edit inventory

### PR DESCRIPTION
**Pull request by murphyad**
Created a modal in order to edit inventory items. Edit button shows only on inventory pages for orgs the user is a part of.  Every property except for units can be updated.

### CODE REVIEW

File: src/queries/EconomicResource/updateEconominResource.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * This cannot update quantity, that must be done by createEconomicEvent
+ *
+ * @author Aaron Murphy <murphyad@msoe.edu>
+ * @date May 9th, 2019
+ */
+
+import { connect } from "react-redux";
+import { graphql, compose } from "react-apollo";
+import gql from "graphql-tag";
+//import { economicResourceInterface } from "./economicResourceInterface"
```
Comment (larsonme) [inline src/queries/EconomicResource/updateEconominResource.tsx]: Remove this commented out part.  Also, the filename is misspelled

File: src/pages/Organizations/organizationInventory.js [hunk 2]
```diff
@@ -28,15 +31,52 @@ const AddItemButton = getMyAgent(({agent, loading, error}) => {
...
         }
         return <div/>
     }
 
 });
 
+class AddItemButton extends React.Component {
+
+    createItem = () => {
+        history.push("/CreateInventoryItem/"+orgId);
```
Comment (larsonme) [inline src/pages/Organizations/organizationInventory.js]: Try merging dev down into here.  We will want the modal changes in here before we merge I believe

**Comment by larsonme** [comment]
2 more things I found:
1) Use trim() on the name of the item before saving and don't let the user save it if they just put in whitespace for the name.  By using trim(), it will make an all whitespace string an empty string
2) Don't let the user save negative values for the quantity. Probably should be added to creating inventory item also, but thats an inconsistency we will put in the documentation for next year

**Review by DonalMoloneyMSOE — APPROVED** [review]
Would it be possible to turn the field that displays the unit ex: sheet, etc and make it a label instead of a textbox? As it is not changeable. Also it appears that the input for note and the input for where the units are different hues. **Notes** text color is a lot stronger than **sheet**. Also maybe for the buttons have them called **Save Changes** and **Upload new Img**, just to be 100% sure the user will now what buttons do.
EditImg

**Comment by larsonme** [comment]
> Would it be possible to turn the field that displays the unit ex: sheet, etc and make it a label instead of a textbox? As it is not changeable. Also it appears that the input for note and the input for where the units are different hues. **Notes** text color is a lot stronger than **sheet**. Also maybe for the buttons have them called **Save Changes** and **Upload new Img**, just to be 100% sure the user will now what buttons do.
> EditImg

@DonalMoloneyMSOE , I'll throw these changes in before we merge tonight to help aaron out.  I'll update the button names, but other than that keep the other stuff the same since it isn't that big of a deal and we are running out of time

**Review by larsonme — APPROVED** [review]

File: src/pages/Organizations/organizationInventory.js [hunk 3]
```diff
@@ -28,15 +32,40 @@ const AddItemButton = getMyAgent(({agent, loading, error}) => {
...
+                connected = true;
             }
-
         }
         return <div/>
     }
 
 });
 
+class AddItemButton extends React.Component {
```
Comment (StrikeNP) [inline src/pages/Organizations/organizationInventory.js]: Why are these whole classes?

**Review by StrikeNP — COMMENTED** [review]
## Description
If I try update the Notes of an item that has notes in it to "" then it returns success but dosn't update. 

## Steps to reproduce
Notes = "My preexisting note"
*User updates notes to be "" (empty string)*

## Expected result:
Notes = ""

## Actual result:
Notes = "My preexisting note"

File: src/pages/Organizations/organizationInventory.js (hunk 3 above)
Comment (larsonme) [reply src/pages/Organizations/organizationInventory.js]: Removed

Comment (larsonme) [reply src/pages/Organizations/organizationInventory.js]: Also resolved your empty string issue with a commit(not proud of it, but it works with graphql)

**Comment by StrikeNP** [comment]
If a user goes to update an item but accidentally deletes the quantity field, it will remove the item from their inventory. Deleting items and data should be done very carefully, and thus this is a bug. This will be made into a  bug report as this issue shouldn't be large enough to prevent presentation to Bob and Lynn.

**Pull request merged**